tests/macsec/test_static_key_ciphers.py

```diff
@@ -0,0 +1,132 @@
+    if request.config.getoption("neighbor_type") != "sonic":
+        pytest.skip("Neighbor type must be sonic")
+    dut_asn = tbinfo['topo']['properties']['configuration_properties']['common']['dut_asn']
+    asic_index = random.choice(duthost.get_frontend_asic_ids())
+    logger.debug(f"ASIC index: {asic_index}")
+    skip_hosts = duthost.get_asic_namespace_list()
+    if duthost.is_multi_asic:
+        cli_options = " -n " + duthost.get_namespace_from_asic_id(asic_index)
+    else:
+        cli_options = ''
+    dut_lldp_table = duthost.shell("show lldp table")['stdout'].split("\n")[3].split()
+    dut_to_neigh_int = dut_lldp_table[0]
+    neigh_to_dut_int = dut_lldp_table[4]
+    neigh_name = dut_lldp_table[1]
+    neighhost = nbrhosts[dut_lldp_table[1]]["host"]
+    if neighhost.is_multi_asic:
+        neigh_cli_options = " -n " + neighhost.get_namespace_from_asic_id(neighhost.get_frontend_asic_ids())
+    else:
+        neigh_cli_options = ''
+    logger.debug("dut cli: {} neigh cli: {}".format(cli_options, neigh_cli_options))
+
+    int_list = {
+        dut_to_neigh_int: {
+            'host': neighhost,
+            'port': neigh_to_dut_int
+        }
+    }
+    time.sleep(45)
+
+    # gather IP address information
+    dut_ip_v4 = tbinfo['topo']['properties']['configuration'][neigh_name]['bgp']['peers'][dut_asn][0]
+    dut_ip_v6 = tbinfo['topo']['properties']['configuration'][neigh_name]['bgp']['peers'][dut_asn][1].lower()
+    bgp_facts = duthost.bgp_facts(instance_id=asic_index)['ansible_facts']
+    for k, v in bgp_facts['bgp_neighbors'].items():
+        if v['description'].lower() not in skip_hosts:
+            if v['description'] == neigh_name:
+                if v['ip_version'] == 4:
+                    neigh_ip_v4 = k
+                elif v['ip_version'] == 6:
+                    neigh_ip_v6 = k
+                logger.debug(v['state'])
+                assert v['state'] == 'established'
+
+    with open(os.path.dirname(__file__) + '/profile.json') as f:
+        macsec_profiles = json.load(f)
+
+    cleanup_macsec_configuration(duthost, ctrl_links, profile_name)
+
+    # wait to ensure link has come up with no macsec
+    time.sleep(45)
+    macsec_check(duthost, cli_options, dut_to_neigh_int, neigh_ip_v4, neigh_ip_v6, macsec=False)
+
+    # use each macsec profile and verify operation
+    for k, v in list(macsec_profiles.items()):
+        if duthost.facts["asic_type"] == "vs" and v['send_sci'] == "false":
+            # On EOS, portchannel mac is not same as the member port mac (being as SCI),
+            # then src mac is not equal to SCI in its sending packet. The receiver of vSONIC
+            # will drop it for macsec kernel module does not correctly handle it.
+            continue
+        else:
+            logger.debug("k: {} v: {}".format(k, v))
+            setup_macsec_configuration(duthost, int_list, k, v['priority'],
```

Comment:
how can it work if we only change the MACsec profile of the duthost without updating the neighbor's profile? At least, the CAK key will be different. Isn't that?

Reply:
The setup_macsec_configuration function in use here configured both DUT and the neighbor.